solr/core/src/test/org/apache/solr/core/CoreSorterTest.java

```diff
@@ -141,12 +142,15 @@ public void integrationTest() {
       }
       @SuppressWarnings({"unchecked"})
       DocCollection col =
-          new DocCollection(
+          DocCollection.create(
               collection,
               sliceMap,
               Collections.singletonMap(
                   ZkStateReader.CONFIGNAME_PROP, ConfigSetsHandler.DEFAULT_CONFIGSET_NAME),
-              DocRouter.DEFAULT);
+              DocRouter.DEFAULT,
```

Comment:
just out of curiosity, since I see the same pattern replaced over and over in this PR, is there a cleaner way to deal with the three new parameters to the `.create` method?